eb_hooks.py

```diff
@@ -371,13 +371,17 @@ def parse_hook_zen4_module_only(ec, eprefix):
         (ecname in ['foss', 'gompi'] and ecversion == '2022b') or
         (ecname in ['GCC', 'GCCcore'] and LooseVersion(ecversion) == LooseVersion('12.2.0'))
     ):
+        env_varname="EESSI_IGNORE_LMOD_ERROR_ZEN4_GCC1220"
+        # TODO: I need to think about how to unset this, at the end of the build for this module
```

Comment:
TBH do you really need to worry about unsetting it? It only affects the modules that reference it, and that's what you want anyway.

The `build_options` do need to be restored though, but there is probably a certain step where you can do that (when the module is created seems like the right choice?)